jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java

```diff
@@ -81,9 +85,27 @@ public static void registerProcessor(VariableProcessor processor) {
         variablesSummary.addAll(ofNullable(filteredVariables).orElse(emptyList())
                                         .stream()
                                         .map(new ProcessInstanceVariableMapper(deploymentId, serverTemplateId, ""))
-                                        .collect(toList()));
-
-        variablesSummary.sort(Comparator.comparing(ProcessVariableSummary::getName));
+                                        .collect(Collectors.toList()));
+
+        Comparator<ProcessVariableSummary> comparator;
+        switch (ofNullable(sortBy).orElse(COL_PROCESS_INSTANCE_VAR_ID)) {
+            case COL_PROCESS_INSTANCE_VAR_ID:
+                comparator = comparing(ProcessVariableSummary::getVariableId);
+                break;
+            case COL_PROCESS_INSTANCE_VAR_VALUE:
+                comparator = comparing(ProcessVariableSummary::getNewValue);
+                break;
+            case COL_PROCESS_INSTANCE_VAR_TYPE:
+                comparator = comparing(ProcessVariableSummary::getType);
+                break;
+            case COL_PROCESS_INSTANCE_VAR_LASTMOD:
+                comparator = comparing(ProcessVariableSummary::getTimestamp);
+                break;
+            default:
+                comparator = comparing(ProcessVariableSummary::getName);
+        }
+
+        variablesSummary.sort(!asc ? comparator : comparator.reversed());
```

Comment:
Isn't the logic reversed here? I'd assume default order for Comparator<T> is ascending (relying on compareTo() implementation of the types passed to comparing()) .... so when asc is true, I'd expect sorting using the comparator, when false by its reversed counterpart.